worlds/lufia2ac/__init__.py

```diff
@@ -0,0 +1,351 @@
+
+    def create_items(self) -> None:
+        item_pool: List[str] = \
+            self.multiworld.random.choices(sorted(self.item_name_groups["Blue chest items"]), k=self.blue_chest_count)
+        if self.capsule_monsters_available == CapsuleMonstersAvailable.option_shuffled_into_multiworld:
+            item_pool += self.item_name_groups["Capsule monsters"]
+            self.blue_chest_count += len(self.item_name_groups["Capsule monsters"])
+        if self.party_members_available == PartyMembersAvailable.option_shuffled_into_multiworld:
+            item_pool += self.item_name_groups["Party members"]
+            self.blue_chest_count += len(self.item_name_groups["Party members"])
+        for item_name in item_pool:
+            item_data: ItemData = l2ac_item_table[item_name]
+            item_id: int = items_start_id + item_data.code
+            self.multiworld.itempool.append(L2ACItem(item_name, item_data.classification, item_id, self.player))
+
+    def set_rules(self) -> None:
+        for i in range(1, self.blue_chest_count):
+            if i % CHESTS_PER_SPHERE == 0:
+                set_rule(self.multiworld.get_location(f"Blue chest {i + 1}", self.player),
+                         lambda state, j=i: state.has("Progressive chest access", self.player, j // CHESTS_PER_SPHERE))
+                set_rule(self.multiworld.get_location(f"Chest access {i + 1}-{i + CHESTS_PER_SPHERE}", self.player),
+                         lambda state, j=i: state.can_reach(f"Blue chest {j}", "Location", self.player))
+            else:
+                set_rule(self.multiworld.get_location(f"Blue chest {i + 1}", self.player),
+                         lambda state, j=i: state.can_reach(f"Blue chest {j}", "Location", self.player))
+
+        set_rule(self.multiworld.get_entrance("FinalFloorEntrance", self.player),
+                 lambda state: state.can_reach(f"Blue chest {self.blue_chest_count}", "Location", self.player))
+        set_rule(self.multiworld.get_location("Iris Treasures", self.player),
+                 lambda state: state.can_reach(f"Blue chest {self.blue_chest_count}", "Location", self.player))
+        set_rule(self.multiworld.get_location("Boss", self.player),
+                 lambda state: state.can_reach(f"Blue chest {self.blue_chest_count}", "Location", self.player))
+        if self.capsule_monsters_available == CapsuleMonstersAvailable.option_shuffled_into_multiworld:
+            add_rule(self.multiworld.get_location("Boss", self.player), lambda state: state.has("DARBI", self.player))
+        if self.party_members_available == PartyMembersAvailable.option_shuffled_into_multiworld:
+            add_rule(self.multiworld.get_location("Boss", self.player), lambda state: state.has("Dekar", self.player)
+                     and state.has("Guy", self.player) and state.has("Arty", self.player))
+
+        if self.goal == Goal.option_final_floor:
+            self.multiworld.completion_condition[self.player] = \
```

Comment:
not important, but it'd be nice if this was an event location and item for the spoiler